bin/await

```diff
@@ -2,217 +2,100 @@
 
 set -euo pipefail
 
-main() {
-
-	local subcommand="${1:-}"
-	shift
-
-	case "${subcommand}" in
-		http)
-			local counter=0 retry=3
+usage() {
+	cat <<'EOF'
+
+USAGE: await [-r RETRIES] [ URI | "cmd" ] [ -- EXTRA_OPTS ]
+
+Forms:
+
+	await [ -r RETRIES ] http://host:port -- [ CURL_OPTS ]
+	await [ -r RETRIES ] https://host:port -- [ CURL_OPTS ]
+	await [ -r RETRIES ] mongodb://host:port -- [ -t SELECTION_TIMEOUT ]
+	await [ -r RETRIES ] redis://host:port -- [ -t CONNECTION_TIMEOUT ]
+	await [ -r RETRIES ] memcached://host:port -- [ -t CONNECTION_TIMEOUT ]
+	await [ -r RETRIES ] mysql://user:password@host:port -- [ -t CONNECTION_TIMEOUT ]
+	await [ -r RETRIES ] dynamodb://host:port -- [ -t CONNECTION_TIMEOUT ]
+	await [ -r RETRIES ] cmd -- COMMAND [ OPTIONS ]
+EOF
+}
 
-			while getopts ":r:" opt; do
-				case "${opt}" in
-					r)
-						retry="${OPTARG}"
-						;;
-					\?)
-						echo "Invalid option: -${OPTARG}" 1>&2
-						return 1
-						;;
-					:)
-						echo "Option -${OPTARG} requires an argument." 1>&2
-						return 1
-						;;
-				esac
-			done
+await() {
+	local retries="${1}" counter=0
```

Comment:
I would add that in `main` where user validation happens; not inside private functions.